pwdtk/middlewares.py

```diff
@@ -38,58 +33,26 @@ def __init__(self, get_response=None):
         else:
             super(PwdtkMiddleware, self).__init__()
 
-    def must_renew_password(self, request):
-
-        if request.method == 'GET':
-            return False
-
-        if django.VERSION < (1, 10):
-            is_authenticated = request.user and request.user.is_authenticated()
-        else:
-            is_authenticated = request.user and request.user.is_authenticated
-
-        if not is_authenticated:
-            return False
-
-        if not PwdData.get_or_create_for_user(request.user).must_renew:
-            return False
-
-        if request.path == reverse(PwdtkSettings.PWDTK_PASSWD_CHANGE_VIEW):
-            return False
-
-        if request.path in PwdtkSettings.PWDTK_PASSWD_CHANGE_ALLOWED_PATHS:
-            return False
-
-        return True
+    def process_request(self, request):
+        if hasattr(PwdtkSettings, "reset_cache"):
```

Comment:
doc string?

what `cache` will be reset?

what is the cache for and why do we have to reset it sometimes

Reply:
shit, resolved dont save comments

Reply:
Te goal is to allow live pwdtk settings. Sometimes setting could come from other sources like DB entries.
We keep a cache only for one request instance, thsi allow change parameters or activate/deactivate pwdtk whit not restart required